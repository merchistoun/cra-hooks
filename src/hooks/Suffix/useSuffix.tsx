export interface UseSuffix {
  addSuffix: (value: number) => string;
}

export const useSuffix = (): UseSuffix => {
  const addSuffix = (value: number): string => {
    switch (value % 10) {
      case 1:
        return `${value}st`;
      case 2:
        return `${value}nd`;
      case 3:
        return `${value}rd`;
      default:
        return `${value}th`;
    }
  };

  return { addSuffix };
};
