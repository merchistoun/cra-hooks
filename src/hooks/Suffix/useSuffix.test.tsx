import { UseSuffix, useSuffix } from "./useSuffix";
import { renderHook } from "@testing-library/react";

describe("useAddSuffix hook", () => {
  it("should mount without error", async () => {
    let hook: UseSuffix;

    const { result } = renderHook(() => useSuffix());
    hook = result.current;

    expect(hook).toMatchSnapshot();

    const { addSuffix } = hook;
    expect(addSuffix(1)).toBe("1st");
    expect(addSuffix(2)).toBe("2nd");
    expect(addSuffix(3)).toBe("3rd");
    expect(addSuffix(4)).toBe("4th");
    expect(addSuffix(5)).toBe("5th");
    expect(addSuffix(1000000)).toBe("1000000th");
  });
});
