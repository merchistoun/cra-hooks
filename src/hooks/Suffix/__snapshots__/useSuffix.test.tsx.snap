// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useAddSuffix hook should mount without error 1`] = `
Object {
  "addSuffix": [Function],
}
`;
